### ISSUE TITLE
Python3 AGI: write() argument must be str, not bytes

### ISSUE THREAD

**Issue by marrold** [rdegges/pyst2#40]
When using asterisk.agi with python 3, I get the following exception

`write() argument must be str, not bytes`

Does anyone have any ideas how to fix this?

Thanks

**Comment by evilscientress** [comment]
I have the same error. Removing the if clause in lines 166-169 and replacing it with `self.stdout.write(command)` seams like the correct fix for me.

I had this working with Python 3.4, and TBH I don't know why those lines ever worked in the first place, because in no version of Python 3 `sys.stdout.write()` accepted bytes, unless you call `sys.stdout.detach()` first.

Refs:
-  [sys.stdout documentation](https://docs.python.org/3/library/sys.html#sys.stdout)

**Comment by angelelena** [comment]
With debian 11 and python 3.9, may be:

```
def send_command(self, command, *args):
        """Send a command to Asterisk"""
        command = command.strip()
        command = '%s %s' % (command, ' '.join(map(str, args)))
        command = command.strip()
        if command[-1] != '\n':
            command += '\n'
        self.stderr.write('    COMMAND: %s' % command)
        self.stdout.write(command.encode('utf8'))
        self.stdout.flush()
```  

Basically: 

```
self.stdout.write(command.encode('utf8'))
```

**Comment by PraveenChordia** [comment]
you need to send data to write in bytes instead of str
```

def send_command(self, command, *args):
  """Send a command to Asterisk"""
  command = command.strip()
  command = '%s %s' % (command, ' '.join(map(str, args)))
  command = command.strip()
  if command[-1] != '\n':
      command += '\n'
  if type(command) is str:
      command = command.encode('utf8')
  self.stderr.write('    COMMAND: %s' % command)
  self.stdout.write(command)
  self.stdout.flush()

```

this worked for me
